src/System.Runtime.Extensions/tests/System/IO/PathTests.cs

```diff
@@ -932,6 +932,173 @@ public static void GetFullPath_Windows_Wildcards(char wildcard)
             AssertExtensions.Throws<ArgumentException>("path", null, () => Path.GetFullPath("test" + wildcard + "ing"));
         }
 
+        [Fact]
+        [PlatformSpecific(TestPlatforms.Windows)]
+        public static void GetFullPath_BasePath_InvalidInput()
+        {
+            Assert.Throws<ArgumentNullException>(() => Path.GetFullPath("", null));
+            Assert.Throws<ArgumentNullException>(() => Path.GetFullPath("tmp", null));
+            Assert.Throws<ArgumentNullException>(() => Path.GetFullPath(@"c:\git\corefx", null));
+            Assert.Throws<ArgumentException>(() => Path.GetFullPath("", @"foo\bar"));
+            Assert.Throws<ArgumentException>(() => Path.GetFullPath("tmp", @"foo\bar"));
+            Assert.Throws<ArgumentException>(() => Path.GetFullPath(@"C:\git", @"foo\bar"));
+        }
+
+        public static IEnumerable<object[]> GetFullPath_BasePath_BasicExpansions_TestData()
+        {
+            string curDir = @"c:\git\corefx";
+            yield return new object[] { curDir, curDir, curDir };
+            yield return new object[] { "", curDir, curDir };
+            yield return new object[] { "..", curDir, Path.GetDirectoryName(curDir) };
+            yield return new object[] { Path.Combine(curDir, ".", ".", ".", ".", "."), curDir, curDir };
+            yield return new object[] { curDir + new string(Path.DirectorySeparatorChar, 3) + ".", curDir, curDir };
+            yield return new object[] { Path.Combine(curDir, "..", Path.GetFileName(curDir), ".", "..", Path.GetFileName(curDir)), curDir, curDir };
+            yield return new object[] { Path.Combine(Path.GetPathRoot(curDir), "somedir", ".."), curDir, Path.GetPathRoot(curDir) };
+            yield return new object[] { Path.Combine(Path.GetPathRoot(curDir), "."), curDir, Path.GetPathRoot(curDir) };
+            yield return new object[] { Path.Combine(Path.GetPathRoot(curDir), "..", "..", "..", ".."), curDir, Path.GetPathRoot(curDir) }; 
+            yield return new object[] { Path.GetPathRoot(curDir) + new string(Path.DirectorySeparatorChar, 3), curDir, Path.GetPathRoot(curDir) };
+
+            // Device paths
+            yield return new object[] { "foo", @"\\?\C:\ ", @"\\?\C:\ \foo" };
```

Comment:
You should have tests with the path with relative segments. That is the most likely scenario, not the basePath. In MSBuild, for example, paths are stored as "..\Utility\Utility.csproj". They would want to combine that with whatever the project root is. Something like "C:\repos\mysolution\coolapp". Right now they set the current directory to do so.